components/Form.js

```diff
@@ -1,10 +1,21 @@
 import styled from "styled-components";
 import categories from "@/assets/categories.json";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 
-export default function CreateForm({ onCreateItem }) {
+export default function CreateForm({
```

Comment:
The component is currently named "CreateForm", but it now handles both creating and editing shopping items.
A more appropriate name could be something like "ShoppingItemForm" to reflect its dual purpose more accurately.

```suggestion
export default function ShoppingItemForm({
```